### PR TITLE
Use transport.request in tests.

### PR DESCRIPTION
**Pull request by dhermes**
In the process
- "spring clean" the modules that were touched
- use `HttpMock` when `HttpMockSequence` not needed
- add some verifications on new `HttpMock`'s

### CODE REVIEW

File: tests/test_jwt.py [hunk 1]
```diff
@@ -19,14 +19,16 @@
...
 
 import mock
+from six.moves import http_client
 import unittest2
 
 from oauth2client import _helpers
 from oauth2client import client
 from oauth2client import crypt
-from oauth2client import file
+from oauth2client import file as file_mod
```
Comment (nathanielmanistaatgoogle) [inline tests/test_jwt.py]: 👎  to abbreviations. `file_module` is just fine.

**Comment by nathanielmanistaatgoogle** [comment]
Looks nearly good to me.

File: tests/test_file.py [hunk 2]
```diff
@@ -29,12 +26,14 @@
...
 import six
 from six.moves import http_client
+from six.moves import urllib_parse
 import unittest2
 
 from oauth2client import _helpers
 from oauth2client import client
-from oauth2client import file
-from .http_mock import HttpMockSequence
+from oauth2client import file as file_mod
```
Comment (dhermes) [inline tests/test_file.py]: @nathanielmanistaatgoogle Doing the rename here

**Comment by dhermes** [comment]
@nathanielmanistaatgoogle I made your `file_mod` -> `file_module` change and tests passed so I merged. If your "nearly LGTM" didn't become "LGTM" after that change, my bad for the merge (I made an assumption).

**Comment by nathanielmanistaatgoogle** [comment]
We're all good here. 👍

**Pull request merged**